sdk/monitor/monitor-query/src/internal/modelConverters.ts

```diff
@@ -102,9 +104,63 @@ export function convertResponseForQueryBatch(
       }))
   };
 
+  (newResponse as any)["__fixApplied"] = fixApplied;
   return newResponse;
 }
 
+/**
+ * This is a workaround for a service bug that we're investigating. The 'body' column will occasionally come
+ * back as a JSON string, instead of being a JSON object.
+ *
+ * (examples, with excess stuff trimmed)
+ * Correct: `{"responses":[{"body":{"tables":[{"name":"PrimaryResult","columns":[{"name":"stringcolumn","type":"string"}],"rows":[["hello"]}`
+ * Broken: `{"responses":[{"body":"{\"tables\":[{\"name\":\"PrimaryResult\",\"columns\":[{\"name\":\"stringcolumn\",\"type\":\"string\"}],\"rows\":[[\"hello\"]}`
+ *
+ * Issue here: https://github.com/Azure/azure-sdk-for-js/issues/15688
+ *
+ * @internal
+ */
+export function fixInvalidBatchQueryResponse(
+  generatedResponse: GeneratedQueryBatchResponse
+): boolean {
+  if (generatedResponse.responses == null) {
+    return false;
+  }
+
+  let wholeResponse: GeneratedQueryBatchResponse | undefined;
+  let hadToFix = false;
+
+  // fix whichever responses are in this broken state (each query has it's own
+  // response, so they're not all always broken)
+  for (let i = 0; i < generatedResponse.responses.length; ++i) {
+    if (
+      generatedResponse.responses[i].body?.tables != null ||
```

Comment:
Is it always the case where both are well formed or neither is? Just wondering if that || should be &&

Reply:
I'm not sure - the problem here is that since it's a service-side bug the behavior is not as well defined as I'd like (ie, I can't force this to happen, it just _does_ sometimes). The only obvious case I've seen fail is the tables one, but (if we start seeing intermittent failures on the errors case, which I need to add with our live testing) we can start tracking it down.

What do you think?

Reply:
In Richard we trust! ™️